### PR TITLE
Do not require space before open parenthesis in a range expression (#3040)

### LINKED ISSUE #3040: SA1008 is shown for range expressions
The code below produces SA1008 "Opening parenthesis should be preceded by space" which I think shouldn't happen because double dots already give enough leeway for readability:

```csharp
return output[..(finalOutputLen - 1)];
```

With a preceding space, it just looks like a misformatted source code.

StyleCop Version: 1.1.118, .netstandard2.1, .NET Core 3.0

### PR DESCRIPTION
**Pull request by ssg**
This is my proposal to fix #3040 . Some troubles I had:

- C# 8.0 code emits compiler errors so I had to add them to the list of expected diagnostic results which isn't ideal.
- I didn't add tests for "fixed code" tests. I wasn't sure if I was supposed to.

### CODE REVIEW

File: StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp8/SpacingRules/SA1008CSharp8UnitTests.cs [hunk 1]
```diff
@@ -3,9 +3,59 @@
...
+            return str[..(finalLen - 1)];
+        }
+    }
+}
+";
+
+            DiagnosticResult[] expectedDiagnostic = new[]
+            {
+                // I couldn't remove these compiler errors, so i expect them in the output
+                // instead.
```
[the remark this answers is not in the document]
Comment (sharwell) [reply StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp8/SpacingRules/SA1008CSharp8UnitTests.cs]: 💭 You might need to define stubs for the `Index` and/or `Range` types so the compiler knows they exist. At worst case something like this:

https://github.com/dotnet/roslyn/blob/1de29ff0ae5661a0ac4b8208ac42570bce4c9417/src/Compilers/Test/Utilities/CSharp/TestSources.cs#L224-L384

But you might be able to remove a bunch of the members and stub the remaining ones with `=> throw null`.

[the remark this answers is not in the document]
Comment (ssg) [reply StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp8/SpacingRules/SA1008CSharp8UnitTests.cs]: Gotcha. Will update it when I get back home.

[the remark this answers is not in the document]
Comment (ssg) [reply StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp8/SpacingRules/SA1008CSharp8UnitTests.cs]: I fixed this (and the tests are passing) but AppVeyor build failed for some other irrelevant test, which I didn't touch: `StyleCop.Analyzers.Test.CSharp7.ReadabilityRules.SA1134CSharp7UnitTests.VerifyMultipleAttribut...` (can't copy the full name on AppVeyor UI).

How can we go forward with this? Is there a way to trigger AppVeyor again without touching the code or do I need to submit a dummy commit?

**Comment by ssg** [comment]
I fixed this (and the tests are passing) but AppVeyor build failed for some other irrelevant test, which I didn't touch: `StyleCop.Analyzers.Test.CSharp7.ReadabilityRules.SA1134CSharp7UnitTests.VerifyMultipleAttribut...` (can't copy the full name on AppVeyor UI).

How can we go forward with this? Is there a way to trigger AppVeyor again without touching the code or do I need to submit a dummy commit?

**Comment by ssg** [comment]
(I triggered it again by removing a frivilous commit)

**Pull request merged**